driver-core/src/main/com/mongodb/internal/connection/SingleServerCluster.java

```diff
@@ -94,7 +95,9 @@ public void serverDescriptionChanged(final ServerDescriptionChangedEvent event)
                     }
                 }
             }
-            publishDescription(descriptionToPublish);
+            if (shouldPublishChangeEvent(event.getNewDescription(), event.getPreviousDescription())) {
```

Comment:
This logic also needs to be applied in `AbstractMultiServerCluster`

Reply:
Done